test/apps/iodemo/io_demo.cc

```diff
@@ -785,13 +786,60 @@ class P2pDemoCommon : public UcxContext {
         return (data_size + chunk_size - 1) / chunk_size;
     }
 
+    static void handle_validate_error(const std::exception &e,
+                                      const UcxConnection *conn) {
+        LOG << "\"" << e.what() << "\" detected on " << conn->get_log_prefix()
+            << " (status=" << ucs_status_string(conn->ucx_status()) << ")";
+        abort();
+    }
+
+    class DataCorruptionError : public std::exception {
```

Comment:
maybe add a generic exception class with std::string member which contains the exception description?
it will be a base class for both DataCheckError and SnMismatchError

Reply:
done
but I use `std::stringstream` for easier string construction